apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx

```diff
@@ -2,48 +2,53 @@ import { ImageIcon } from "~/shared/icons";
 import { Flex, Grid, Heading } from "~/shared/design-system";
 import { useEffect, useState } from "react";
 import { useActionData } from "@remix-run/react";
-import { Asset } from "@webstudio-is/prisma-client";
 import { AssetManagerImage } from "./components/image";
 
 import { AddAnAssetForm } from "./components/add-an-asset-form";
-import { UploadingAsset } from "../types";
+import { Asset } from "../types";
 
-export const TabContent = ({
-  assets: baseAssets,
-}: {
-  assets: Array<Asset | UploadingAsset>;
-}) => {
-  const newImages = useActionData();
+export const useAssetsState = (baseAssets: Array<Asset>) => {
+  const imageChanges = useActionData();
 
-  const [assets, setAssets] = useState(baseAssets);
+  const [assets, setAssets] = useState<Asset[]>(baseAssets);
 
   useEffect(() => {
-    if (newImages?.length) {
+    if (imageChanges?.uploadedAssets?.length) {
       setAssets((currentAssets) => [
```

Comment:
I wonder if there was an actual case for using setState callback here

Reply:
hm?

Reply:
setAsset(callback) vs assets

Reply:
I always tend to use the callback, its way safer

Reply:
by doing so you are waiting for the next update cycle, this is only useful when you know there can be an update that came from somewhere else, its not a thing we need to use as a general pattern, only in special cases

Reply:
removing it actually broke it, this makes sure we are using the latest version

Reply:
because you didn't set the dependency?

Reply:
I did,updated it to include baseAssets